trieste/utils/multi_objectives.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains synthetic multi-objective functions, useful for experimentation.
+"""
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+from functools import partial
+
+import tensorflow as tf
+
+from ..type import TensorType
+
+
+class MultiObjectiveTestProblem(ABC):
+    """
+    Base class for synthetic multi-objective test functions.
+    Prepares the synthetic function and generates pareto optimal points.
+    The latter can be used as a reference of performance measure of certain
+    multi-objective optimization algorithms.
+    """
+
+    @property
+    @abstractmethod
+    def dim(self):
```

Comment:
These should have return types:

```suggestion
    def dim(self) -> int:
```

Reply:
thanks, sorry should add this earlier.
Btw do we also need a return type for `bounds`? is that `ListN` or maybe `SequenceN` ?

Reply:
`ListN`is good (I haven't merged the change to `SequenceN` yet, and I'm happy to wait until this is merged in.)

Reply:
Thanks, aware that `ListN` maybe more of used in test so currently used `list[list[float]]` for bounds.